opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -12,141 +12,135 @@
+
+
+def get_value(key: str, context: typing.Optional[Context] = None) -> "object":
+    """To access the local state of an concern, the RuntimeContext API
+    provides a function which takes a context and a key as input,
+    and returns a value.
+
+    Args:
+        key: The key of the value to retrieve.
+        context: The context from which to retrieve the value, if None, the current context is used.
+    """
+    return context.get(key) if context is not None else get_current().get(key)
+
+
+def set_value(
+    key: str, value: "object", context: typing.Optional[Context] = None
+) -> Context:
+    """To record the local state of a cross-cutting concern, the
+    RuntimeContext API provides a function which takes a context, a
+    key, and a value as input, and returns an updated context
+    which contains the new value.
+
+    Args:
+        key: The key of the entry to set
+        value: The value of the entry to set
+        context: The context to copy, if None, the current context is used
+    """
+    if context is None:
+        context = get_current()
+    new_values = context.copy()
+    new_values[key] = value
+    return Context(new_values)
+
+
+def remove_value(
+    key: str, context: typing.Optional[Context] = None
+) -> Context:
+    """To remove a value, this method returns a new context with the key
+    cleared. Note that the removed value still remains present in the old
+    context.
+
+    Args:
+        key: The key of the entry to remove
+        context: The context to copy, if None, the current context is used
+    """
+    if context is None:
+        context = get_current()
+    new_values = context.copy()
+    new_values.pop(key, None)
+    return Context(new_values)
+
+
+def _init_runtime() -> None:
+    # FIXME use a better implementation of a configuration manager to avoid having
+    # to get configuration values straight from environment variables
+
+    global _RUNTIME_CONTEXT  # pylint: disable=global-statement
+    configured_context = environ.get(
+        "OPENTELEMETRY_CONTEXT", "default_context"
+    )  # type: str
+    try:
+        _RUNTIME_CONTEXT = next(
+            iter_entry_points("opentelemetry_context", configured_context)
+        ).load()()
+    except Exception:  # pylint: disable=broad-except
+        logger.error("Failed to load context: %s", configured_context)
+
+
+def get_current() -> Context:
+    """To access the context associated with program execution,
+    the RuntimeContext API provides a function which takes no arguments
+    and returns a RuntimeContext.
+    """
+
+    if _RUNTIME_CONTEXT is None:
+        _init_runtime()
+
+    return _RUNTIME_CONTEXT.get_current()  # type:ignore
+
+
+def set_current(context: Context) -> Context:
+    """To associate a context with program execution, the Context
+    API provides a function which takes a Context.
+
+    Args:
+        context: The context to use as current.
+    """
+    if _RUNTIME_CONTEXT is None:
```

Comment:
This can be made more uniform by having:

```python
if context is None:
    context = get_current()
```
in all the functions defined in this API. Doing son makes it not necessary to have `_init_runtime`, just move its code into `get_current`.

Reply:
Fixed.